src/main/java/com/jcabi/log/VerboseProcess.java

```diff
@@ -473,6 +487,106 @@ public Void call() throws Exception {
+            int count = 0;
+            while (true) {
+                if (Thread.interrupted()) {
+                    Logger.debug(
+                        VerboseProcess.class,
+                        "explicitly interrupting read from buffer"
+                    );
+                    break;
+                }
+                if (!previous.isEmpty()) {
+                    appendLine(previous, builder);
+                    previous = EMPTY_STRING;
+                }
+                final String line = reader.readLine();
+                if (line == null) {
+                    logAndClear(writer, this.level, builder);
+                    break;
+                }
+                if (shouldAppend(line)
+                        && count < MAX_STACK_LENGTH) {
+                    appendLine(line, builder);
+                    ++count;
+                } else {
+                    if (builder.length() > 0) {
+                        logAndClear(writer, this.level, builder);
+                    }
+                    count = 1;
+                    previous = line;
+                }
+            }
+        }
+
+        /**
+         * Appends a line and a newline character to the builder.
+         * @param line String to append
+         * @param builder StringBuilder with log statement
+         */
+        private static void appendLine(final String line,
+                final StringBuilder builder) {
+            builder.append(line);
+            builder.append(NEW_LINE);
+        }
+
+        /**
+         * Logs StringBuilder to supplied Logger and Writer then clears out
+         * the builder.
+         * @param writer Writer to use
+         * @param level Level to log at
+         * @param builder StringBuilder with log statement
+         * @throws IOException writer could throw this
+         */
+        private static void logAndClear(final BufferedWriter writer,
+                final Level level, final StringBuilder builder)
+                throws IOException {
+            if (builder.length() > 0) {
+                final String text = builder.toString();
+                Logger.log(level, VerboseProcess.class, LOG_FORMAT, text);
+                writer.write(text);
+            }
+            builder.setLength(0);
+        }
+
+        /**
+         * Checks if line is part of a stack trace and should be appended.
+         * @param input String to check
+         * @return Result, true or false
+         */
+        private static boolean shouldAppend(final String input) {
+            final String stripped = stripStart(input);
+            return stripped.startsWith(PREFIX_AT)
+                || stripped.startsWith(PREFIX_CB)
+                || stripped.startsWith(PREFIX_DOTS);
+        }
+
+        /**
+         * Strips whitespace at beginning of String.
+         * @param input String to strip
+         * @return Stripped string
+         */
+        private static String stripStart(final String input) {
```

Comment:
@dean-e-clark this method should be named like a noun, please read http://www.yegor256.com/images/books/elegant-objects/seven-pages.pdf